pkg/kn/commands/plugin/install.go

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2024 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/config"
+)
+
+var updateFlag bool
+
+// NewPluginInstallCommand creates a new `kn plugin install` command
+func NewPluginInstallCommand(p *commands.KnParams) *cobra.Command {
+
+	var localPath string
+	cmd := &cobra.Command{
+		Use:     "install",
+		Short:   "Install a plugin from URL or local file",
+		Aliases: []string{"ls"},
+		Long: `Install a plugin from URL or local filesystem.
+
+Plugin is installed from a provided URL or local path to ${KN_CONFIG_DIR}/kn/plugins.
+
+Current: ` + config.GlobalConfig.PluginsDir(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				if localPath == "" {
+					return errors.New("'kn plugin install' requires URL or local file parameter '-f' to be specified")
```

Comment:
```suggestion
					return errors.New("'kn plugin install' requires the URL as argument or local file specified via '-f'")
```
When I read this error message, I think it might maybe sense to unify this, and just assume a local file if not schema is given in the URL (or when the schema is "file://"). 

wdyt ? That would make the UX also nice and we would need an extra option `-f` for that.